modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/FDSAdminPortlet.java

```diff
@@ -122,22 +98,7 @@ protected void doDispatch(
 			RenderRequest renderRequest, RenderResponse renderResponse)
 		throws IOException, PortletException {
 
-		ThemeDisplay themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
-			WebKeys.THEME_DISPLAY);
-
-		try {
-			BatchEngineUnitThreadLocal.setFileName(_bundle.toString());
-
-			_generate(
-				themeDisplay.getCompanyId(), themeDisplay.getLocale(),
-				themeDisplay.getUserId());
-		}
-		catch (Exception exception) {
-			_log.error(exception);
-		}
-		finally {
-			BatchEngineUnitThreadLocal.setFileName(StringPool.BLANK);
-		}
+		_dataSetModelManager.checkCompany(CompanyThreadLocal.getCompanyId());
```

Comment:
This is just to create data in default company, right? If so, maybe `PortalUtil.getDefaultCompanyId()` is a bit more descriptive.

We are not using `TransactionCommitCallbackUtil` here. Is this intentional?

Reply:
This is a sort of fallback to ensure current company portlet is running at contains the data set object definitions. 99.99% of the time this shall be a no-op check, probably there is no way to reach this code in a state where data model is not yet created, so I'd go to remove this line in the final PR